[4.0] Trigger Finder events

sakiss opened the pull request:
### Summary of Changes
It triggers events after performing the basic finder indexer's actions.
The Indexer performs 3 tasks  (index, delete, purge), though no any event triggered after the Finder Indexer has finished them.

Performing further actions can be very useful in cases like: 
Caching, Meta-data generation, Filters generation, Logging.

### Testing Instructions
Lets add some logic to be executed on each event.
It will be nothing more than the generation of a log record, for each action.

Open the file: plugins/finder/content/content.php

Within the class add/append the following code:
~~~~
      	/**
	 * Triggered after saving an item or after running the indexer.
	 *
	 * @param   Result  $item
	 * @param   Int     $linkid
	 *
	 */
	public function onFinderIndexAfterIndex(Result $item, $linkId)
	{
		\Joomla\CMS\Log\Log::addLogger(array('text_file' => 'finder.log.php'), \Joomla\CMS\Log\Log::ALL, array('finder_plugin'));
		\Joomla\CMS\Log\Log::add('onFinderIndexAfterIndex Run after:' . $linkId, \Joomla\CMS\Log\Log::INFO, 'finder_plugin');
	}

	/**
	 * Triggered after deleting an item or pressing 'Delete' in com_finder.
	 * @param int $linkId
	 *
	 */
	public function onFinderIndexAfterDelete($linkId)
	{
		\Joomla\CMS\Log\Log::addLogger(array('text_file' => 'finder.log.php'), \Joomla\CMS\Log\Log::ALL, array('finder_plugin'));
		\Joomla\CMS\Log\Log::add('onFinderIndexAfterDelete Run after:' . $linkId, \Joomla\CMS\Log\Log::INFO, 'finder_plugin');
	}

	/**
	 * Triggered after purging (press 'Clear Index') in com_finder.
	 */
	public function onFinderIndexAfterPurge()
	{
		\Joomla\CMS\Log\Log::addLogger(array('text_file' => 'finder.log.php'), \Joomla\CMS\Log\Log::ALL, array('finder_plugin'));
		\Joomla\CMS\Log\Log::add('onFinderIndexAfterPurge Run after', \Joomla\CMS\Log\Log::INFO, 'finder_plugin');
	}
~~~~

### Expected result
After appending the above code, log events will be created for each Indexer's actions.
You can check the log records in the file: **administrator/logs/finder.log.php**

1. **Saving** an existing article in com_content, 2 log records should be created:
a. One with the message: _onFinderIndexAfterDelete Run after_ 
b. One with the message: _onFinderIndexAfterIndex Run after_ 

2. **Pressing the "Clear Index"** in the com_finder, 1 log record should be created with the message: _onFinderIndexAfterPurge_

3. **Deleting** an item in com_finder, 1 log record should be created, with the message: _onFinderIndexAfterDelete Run after_

4. **Pressing the "Index"** in the com_finder, multiple log records (as many as the indexed items) should be created with the message: _onFinderIndexAfterIndex Run after_

*As it is clear all the above is just for testing purpose.

### Documentation Changes Required
Not sure

Commit:
Some cleanups

brianteeman committed:
[4.0] Move Private Messages to Users

As the component menu is now used for components that create some form of site content this Pr moves the private messages component from the Component menu to the users menu.

When testing simply applying the patch will only add the messages menu items to the users menu. It will not remove it from the component menu. As the sql change is data and not structure you cannot use the database fix either. It will have to be a direct update in the sql using phpmyadmin or similar

zero-24 committed:
Merge branch '4.0-dev' into messages

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/cleanup/plg_system_cache

Commit:
Load document into application when debug is enabled

Commit:
Merge branch '4.0-dev' into j4/cleanup/plg_system_cache

brianteeman committed:
Merge remote-tracking branch 'origin/4.0-dev' into messages

brianteeman committed:
Merge branch '4.0-dev' into messages

Commit:
Merge branch '4.0-dev' into messages

brianteeman committed:
Update installation/sql/mysql/joomla.sql

Co-Authored-By: Quy <quy@fluxbb.org>

brianteeman committed:
Update installation/sql/mysql/joomla.sql

Co-Authored-By: Quy <quy@fluxbb.org>

brianteeman committed:
Update installation/sql/postgresql/joomla.sql

Co-Authored-By: Quy <quy@fluxbb.org>

brianteeman committed:
Update installation/sql/postgresql/joomla.sql

Co-Authored-By: Quy <quy@fluxbb.org>

brianteeman committed:
Merge branch '4.0-dev' into messages

brianteeman committed:
Merge branch '4.0-dev' into messages

brianteeman committed:
alternate preset thank @infograf768

Commit:
Merge branch '4.0-dev' into j4/cleanup/plg_system_cache

brianteeman committed:
[4.0]

This may be personal preference but I found the background on the "add module to dashboard" too "in your face".

With this PR you can now see the background pattern etc

Commit:
Merge branch '4.0-dev' into j4/cleanup/plg_system_cache

brianteeman committed:
simplified

chmst committed:
Remove unused method getsortfields from views of com_fields, categories, langueges

Commit:
Query cleanup

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/prepared/com-content/frontend

Commit:
Query cleanup

Commit:
Parameterized queries

Commit:
Query cleanup

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/prepared/com-content/frontend

Commit:
CS

Commit:
Parameterized queries

Commit:
Parameterized queries

Commit:
Password toggler markup and styling

Commit:
Merge branch '4.0-dev' into j4/password-view

Commit:
Correct input method

Commit:
Add missing bind

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/prepared/com-content/frontend

Commit:
Back to old icons

Commit:
Base code on input's type

Commit:
Use one button class

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/prepared/com-content/frontend

Commit:
CS

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/password-view

Commit:
Quote columns

brianteeman committed:
Merge branch '4.0-dev' into add_module

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/password-view

brianteeman committed:
[4.0] Dashboards

Before this PR In the dashboards of joomla 4 we have a situation where sometimes there is a large space between _modules_

This is because the _modules_ generated by the system are in a different div than that of the more traditional modules.

By moving them all into the same div that problem disappears.

Commit:
Fix fatal error in HTMLHelper

Commit:
Use service key

Commit:
Merge remote-tracking branch 'upstream/4.0-dev' into j4/password-view

Commit:
Correct styling

Commit:
Use FA icon in postinstall messages

Commit:
Correct TFA field markup as well

Commit:
aria-hidden

Commit:
aria-hidden

richard67 committed:
Merge remote-tracking branch 'upstream/4.0-dev' into messages

richard67 committed:
Dummy change to be removed later

richard67 committed:
Undo dummy change

brianteeman committed:
Merge branch 'richard67-messages' into messages

richard67 committed:
Fix menu lft rgt (#96)

richard67 committed:
Fix where clause in update sql scripts (#97)

brianteeman committed:
Update administrator/components/com_admin/sql/updates/mysql/4.0.0-2019-11-19.sql

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
Update administrator/components/com_admin/sql/updates/postgresql/4.0.0-2019-11-19.sql

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

sakiss committed:
[4.0] Trigger events after any Finder action (index, delete, purge)

sakiss committed:
[4.0] Trigger events after any Finder action (index, delete, purge)

sakiss committed:
Fix code style errors

sakiss committed:
Fix code style errors

mbabker commented on plugins/finder/contacts/contacts.php:
```diff
@@ -384,7 +385,12 @@ protected function index(Result $item)
+
+		if ($linkId)
+		{
+			Factory::getApplication()->triggerEvent('onFinderIndexAfterIndex', array($item));
```
It’s be better to put this trigger in the indexer instead of relying on each plugin to add it.

sakiss replied:
Thanks @mbabker 
It's an abstract method in the Indexer.

sakiss replied:
Thanks!
It is an abstract function in the Indexer

mbabker replied:
Right, but each of the [driver subclasses](https://github.com/joomla/joomla-cms/tree/4.0-dev/administrator/components/com_finder/Indexer/Driver) implements the abstract method, so that would be the better spot for the trigger.

sakiss replied:
Agree it's better there

astridx committed:
[4.0] I deleted a line that I don't think we need. (#27772)

* tidyingup

* one more

brianteeman committed:
[4.0] Update Tinymce (#27520)

* [4.0] Update Tinymce

TinyMce follows semantic versioning. There is no reason that we should be locked down to an old patch release of 5.0.14 when the current is 5.1.5

Some help fixing the javascript in build\build-modules-js\init.es6.js to write the correct version number in the plugin xml would be appreciated

* Fix TinyMCE version in manifest file

* Fix CodeMirror version in manifest file

* Simplify version replacement

* xml

Co-authored-by: SharkyKZ <sharkykz@gmail.com>

brianteeman committed:
[4.0] Obvious typo (#27811)

As you can see the folder name has a typo - this PR fixes that typo AND fixes the reference in the code so the typo doesn't have to bubble down

Hackwar committed:
[4.0] Changing UI for com_mails list view (#27704)

* Changing UI for com_mails list view

* Adding column to com_mails with missing language templates

* Update administrator/components/com_mails/tmpl/templates/default.php

Co-Authored-By: Quy <quy@fluxbb.org>

* Update administrator/components/com_mails/tmpl/templates/default.php

Co-Authored-By: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

rdeutz committed:
[4.0]Simplified -  add module to dashboard background (#27468)

* [4.0]

This may be personal preference but I found the background on the "add module to dashboard" too "in your face".

With this PR you can now see the background pattern etc

* simplified

rdeutz committed:
Merge pull request #27733 from SharkyKZ/j4/password-view

[4.0] Password toggler markup and styling

brianteeman committed:
[4.0]Inconsistent hover (#27803)

Pull request for #25869

Go to the Admin Login page
Hover over the top-right box
OR
Go to any Admin page and hover over the top right icons

### Expected result
Icon and site name change to white at the same time

### Actual result
Text changes instantly
Icon has a small animation + delay

### With this PR
Icon and site name change to white at the same time

### Testing
as this is a scss change don't forget to either `npm i` or `node build.js compile-css`

If you have trouble seeing the issue try changing the value from 0.2s to 1.1s

rdeutz committed:
Merge pull request #27166 from SharkyKZ/j4/cleanup/plg_system_cache

[4.0] [plg_system_cache] Fix debug with page caching enabled

rdeutz committed:
Merge pull request #27096 from brianteeman/messages

[4.0] Move Private Messages to Users

rdeutz committed:
Merge pull request #27799 from SharkyKZ/j4/error/htmlhelper

[4.0] Fix fatal error in HTMLHelper

rdeutz committed:
Merge pull request #27721 from SharkyKZ/j4/prepared/com-content/frontend

[4.0] [com_content] Parameterized queries (frontend)

Commit:
Move columns out of list.select state (#27701)

brianteeman committed:
[4.0] media manager duplicate selectors (#27669)

* [4.0] media manager duplicate selectors

the class media-view-icons should only be declared once in the scss file.

this is just a change for maintainability as the compiled css file that is generated is identical before and after this pr

the minified file can be located `media\com_media\css\mediamanager.min.css`

* Update administrator/components/com_media/resources/styles/components/_media-toolbar.scss

Co-Authored-By: Quy <quy@fluxbb.org>

* Update _media-toolbar.scss

Co-authored-by: Quy <quy@fluxbb.org>

sakiss committed:
Remove plugin triggers from plugins and add them to the Indexer/Driver classes

sakiss committed:
Remove plugin triggers from plugins and add them to the Indexer/Driver classes

sakiss committed:
Fix code style again

sakiss committed:
Fix code style again

brianteeman committed:
masonry

sakiss committed:
Fix code style again

sakiss committed:
Fix code style again

brianteeman committed:
hound

brianteeman committed:
conflicts?

travisrisner committed:
give text field focus when password revealed (#27821)

brianteeman committed:
[4.0]  Remove header SVG curve in error page (#27826)

As the SVG curve has been removed from the template it should be removed from the error page as well.

## Steps to reproduce the issue
Go to the front end
Go to a nonexistent page
See error page with curve in header

## Apply PR
Go to the front end
Go to a nonexistent page
See error page with NO curve in header

Quy commented on administrator/components/com_finder/Model/IndexModel.php:
```diff
@@ -41,6 +41,14 @@ class IndexModel extends ListModel
+	 * The event to trigger after purging the data.
+	 *
+	 * @var    string
+	 * @since  2.5
```
```suggestion
	 * @since  __DEPLOY_VERSION__
```

sakiss replied:
@Quy This should not reference the current version?
Why a version placeholder for a class parameter?

Using 4.0 as version seems more correct to me

Quy replied:
When this PR is merged which will likely be in 4.0, it will be replaced with 4.0.

sakiss replied:
Yes the variable will be replaced during the phing build but it will be always replaced in the future, with the current version. Right?

Quy replied:
No. only the initial merged version.

brianteeman committed:
[4.0] Menutype Filters (#27827)

### Steps to reproduce the issue
Create an admin menu
Create at least one item in the new admin menu
Select any menu (site or admin)
Try to change the Site/Admin filter

### Expected result
List of menu items changes

### Actual result
Nothing happens. Unless you deselect the Menu filter as well

### Apply This PR
When you switch between Site and Administrator the menu filter is reset

Subhang23 committed:
fix error of styling issue with save and close button (#27816)

* fix error of styling issue with save and close button

* fixing the position of dropdown button

* ensured breakdown of dropdown in articles component does not occur

* fixed error in persian language

* correcting linting

* fixed error in transitioning from large window to small window in rtl

ravisaxena23 committed:
[4.0 dev] mobile button visible on desktop (#27818)

* removal of gear on dextop

* removal of gear on dextop

* removal of gear on dextop revised

* removal of gear on dextop revised

Fedik committed:
Update javascript, and litle fixes (#99)

Fedik committed:
fix hound (#100)

Fedik committed:
fix hound (#101)

wilsonge committed:
Fix CS

brianteeman committed:
[4.0] Meta Description character count (#27804)

ceford committed:
[4.0] Atum top (#27793)

Fedik committed:
[4.0] WebAsset support for inline assets (#27756)

chmst committed:
Remove unused getSortFields from content views (#27679)

chmst committed:
Remove unused method getsorfields from views of com_users, menus and modules (#27677)

chmst committed:
Remove unused method getSortFields from components- part 1 (#27678)

brianteeman committed:
[4.0] Display units for size and dimensions in com_media (#27819)

* [4.0] Display units for size and dimensions in com_media

PR for #26696

1. Displays the dimension as 123px * 321px instead of 123 x 321
2. Displays the size in KB not bytes and right align

* Update administrator/components/com_media/resources/scripts/components/infobar/infobar.vue

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_media/resources/scripts/components/infobar/infobar.vue

Co-Authored-By: Quy <quy@fluxbb.org>

* spaces

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Quy <quy@fluxbb.org>

richard67 committed:
Merge pull request #27680 from chmst/remove-getsortfields-general

[4.0] Remove getsortfields from views of com_fields, categories, languages

sakiss committed:
Update administrator/components/com_finder/Model/IndexModel.php

Co-Authored-By: Quy <quy@fluxbb.org>

brianteeman committed:
[4.0] Select multiple (#27833)

* [4.0] Select multiple

* Update plugins/system/debug/debug.xml

Co-Authored-By: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

brianteeman committed:
[4.0] Admin modules - edit links (#27843)

* [4.0] Admin modules - titles

* copy paste

brianteeman committed:
[4.0] Remove Meta Keywords (part 1) (#25258)

* [4.0] Remove Meta Keywords (part 1)

Meta Keywords as a meta tag has pretty much been irrelevant for **10 years**

This PR does the following

1. Removes Site Meta Keywords from global config
2. Removes Menu item Meta Keywords
3. Removes per language Meta Keywords
4. Renames the item field Meta Keywords to Keywords
 - We use this as a field with mod_related_items and com_banners

Part 2 of removing meta keywords will
1. Move the Keyword field to a more appropriate place
2. Check that all the keyword fields are used with mod_related_items and com_banners

### Testing
Check that Meta Keyword field is removed from global config and menu items
Check that renamed Keyword field does not produce a meta key in the frontend

* .

* Update language/en-GB/mod_related_items.sys.ini

Co-Authored-By: Quy <quy@fluxbb.org>

* Update Content.php

* Update Content.php

* Update com_banners.ini

Co-authored-by: Quy <quy@fluxbb.org>

tecpromotion committed:
[4.0] Update com_finder.sys.ini (#27840)

* Update com_finder.sys.ini

Deletion of multiple used strings and alphabetical sorting adjusted.

* fix lines

Fedik committed:
[4.0] Fix codemiror, use original css files (#27847)

brianteeman committed:
[4.0] WebAssetManager Exception messages (#27853)

Correct the English for the WebAssetManager's exception messages.

Commit:
Translate com_finder modal titles (#27852)

Fedik committed:
Update grid cpanel layout css (#103)

wilsonge committed:
Set data directly into state rather than faking a proxy into the input object (#27696)

wilsonge committed:
[4.0] Allow extensions to manipulate data in API on save (#27795)

Use it in a concrete way in the categories view

Fedik committed:
WebAsset for installation/error (#27855)

SharkyKZ committed:
[4.0] Type-safe string comparisons in libraries (#27194)

infograf768 committed:
[4.0] Fix Mobile toolbar and filters alignment RTL and LTR (#27841)

Commit:
Allow to set menu params during component installation (#27849)

Commit:
[4.0] Include resize modal for image type only (#27851)

wilsonge committed:
Template style api shouldn't allow modification of the templates property

brianteeman committed:
[4.0] Border color between tabs on hover (#27860)

wilsonge committed:
Simplify code in plugin controller

brianteeman committed:
[4.0] npm  packages semver (#27848)

This PR updates a few packages that clearly follow semver

brianteeman committed:
[4.0] Login page module (#27865)

The sidebar module on the login page is partially hidden on a mobile view. This PR resolves that.

Simple PR no css changes

bembelimen committed:
Remove negative margin (#27862)

brianteeman committed:
[4.0] Newsfeed Definition List (#27825)

The child count of items is using a definition list. This is completely the wrong markup to use as a number is not the description of the term in this use.

This PR changes it to a span **exactly** the same as every other use of show_cat_items

Fedik committed:
[4.0] WebAsset correct user.css dependencies and sorting (#27872)

SharkyKZ committed:
[4.0] [mod_articles_category] Add items sublayout (#24991)

This adds a layout for items to remove duplicate code.

wilsonge committed:
Remove some places where I don't need to review things

brianteeman committed:
[4.0] Unpublish admin modules from dashboard (#27782)

wilsonge committed:
[4.0] Fix page navigation query (#27871)

Fix page navigation links by fixing query to join on the workflow stages table correctly.

infograf768 committed:
[4.0] Missing installation error language string (#27870)

Commit:
Add margin to back-top (#27861)

Commit:
Remove setFieldAttribute `action` on `catid` in com_content, … (#24329)

Since the introduction of the categoryedit form field in the edit 
models/views of com_content, com_contact, com_newsfeeds and com_banners 
(d6673d7), there is no `action` attribute on the `catid` form field and 
thus no need to set it anymore.

wilsonge committed:
[4.0] Fix undefined property when editing stages/transitions (#27874)

brianteeman committed:
[4.0] Center modal on mobile (#27859)

brianteeman committed:
[4.0] Remove Fade effect (#27857)

Faster login
Solves stutter bug #25904
Solves internet explorer bug #26005

wilsonge committed:
Fix icon reference

wilsonge committed:
Fix tinymce version after last commit

SharkyKZ committed:
[4.0] Namespace Content - Confirm Consent plugin (#27641)

* Namespace Content - Confirm Consent plugin

* Revert

* Move addfieldprefix to form

* Update file list in manifest

* Align

richard67 committed:
[4.0] Allow empty metakey values in database tables (#27854)

* Allow empty metakey values in database tables

* Remove meta from keywords in comments

* Use null instead empty string for empty metakey

* Roll back unnecessary change for categories

* For tags do like for categories because varchar not null in past

brianteeman committed:
[4.0] Codemirror update (#27868)

Codemirror was locked at a specific version that is now very old.

This PR changes the setting in package.json so that it follows semver

Codemirror has always been very good at following semver so this should be totally safe.

(the update process also updated tinymce at the same time)

SharkyKZ committed:
Remove zero opacity (#27878)

brianteeman committed:
[4.0] Main menu add new shortcuts (#27788)

* [4.0] Main menu add new shortcuts

The admin menu module has options to show/hide the "add new" shortcuts on the menu. This option was not working. It is now

* Update administrator/modules/mod_menu/tmpl/default_submenu.php

Co-Authored-By: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

Commit:
[4.0] Load presets from component directory (#27869)

* Move component specific presets to component folder

* Move preset language strings from lib to components

* Dynamically load presets from component folders instead of using hardcoded list.

* Name of preset is without ".xml"

* Move language strings use din presets to respective component

brianteeman committed:
[4.0] Internet Explorer message (#27879)

* [4.0] Internet Explorer message

Although you can kind of use the admin with IE it is not supported and will not work 100%. This PR uses css to detect if you are using ie10 or 11 and displays a warning if you are.

To test `npm i` or `node build.js --compile.css`

Make sure that you do not see a message with your normal browser
If you have access to IE10 or 11 then make sure you do see a message

* typo

* important

* cs

* Update joomla.ini

* Update administrator/templates/atum/scss/blocks/_login.scss

Co-Authored-By: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

Commit:
[4.0] Adding frontend edit for com_contact (#24311)

* Adding frontend edit for com_contact

* Revert change in package-lock.json

* Removing description strings

* Adding newline character at end of file (CS)

* Prevent changing language in frontend when item is already associated.

* Adding type hint for preprocessform so it's similar to parent function.

* Redirect to previos page on cancel

* addfieldpath doesn't work anymore in 4.0

* Adjusting to template changes

* Removing tabstate as it is no longer available.

* Codestyle

* wip

* Import the RouteHelper

* cs

* Taking care of changes regarding FontAwesome Icons

* Update administrator/components/com_contact/Service/HTML/Icon.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update components/com_contact/Controller/ContactController.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update components/com_contact/View/Form/HtmlView.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update components/com_contact/View/Form/HtmlView.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update components/com_contact/tmpl/category/default_items.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update components/com_contact/tmpl/contact/default.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_contact/Service/HTML/Icon.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_contact/Service/HTML/Icon.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_contact/Service/HTML/Icon.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_contact/Extension/ContactComponent.php

Co-Authored-By: Richard Fath <richard67@users.noreply.github.com>

* Correct publish up and down check

* PHPCS

Co-authored-by: Quy <quy@fluxbb.org>
Co-authored-by: Astrid  <astridx@users.noreply.github.com>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

SharkyKZ committed:
[4.0] Use layout attribute to set field layout (#26214)

* Use layout to set layout

* Update forms

* Set layout

* Cleanup

* Double label workaround

* Update newly added fields

* Remove extra quote

* Update forms

Co-authored-by: Quy <quy@fluxbb.org>

brianteeman committed:
[4.0] Update the bump script (#27882)

brianteeman committed:
[4.0] Update the build script (#27886)

Subhang23 committed:
fix error of improper padding for close button in rtl in add modules … (#27883)

* fix error of improper padding for close button in rtl in add modules modal

* changed styling

brianteeman committed:
[4.0] Help Links (#27894)

Without continuing the debate about what type of links should or should not be on the help page there can be no argument that the glip link doesnt beling here. It is not a help link and more importantly it is not a public link/resource

SharkyKZ committed:
[4.0] [com_content] Use parent query in backend featured model (#27890)

* Use query from parent model

* Remove unused imports

SharkyKZ committed:
Remove table prefix (#25390)

brianteeman committed:
[4.0] User Menu heaader (#27893)

* [4.0] User Menu heaader

The display of the user name in the user dropdown menu looks a little off due to lack of styling.

This add an icon before the username and the text "signed in as " as a prefix. (The text is as used on github)

There are css changes so npm i or node buuld.js --compile-css

* Update administrator/modules/mod_user/tmpl/default.php

Co-Authored-By: Quy <quy@fluxbb.org>

* Update administrator/templates/atum/scss/blocks/_header.scss

Co-Authored-By: Quy <quy@fluxbb.org>

* Update administrator/modules/mod_user/tmpl/default.php

Co-Authored-By: SharkyKZ <sharkykz@gmail.com>

* Update administrator/language/en-GB/mod_user.ini

Co-Authored-By: SharkyKZ <sharkykz@gmail.com>

Co-authored-by: Quy <quy@fluxbb.org>
Co-authored-by: SharkyKZ <sharkykz@gmail.com>

Commit:
[4.0] Quicktask for custom adminmenu (#27885)

* Merging parameter `menu-quicktask-link` with `menu-quicktask` as we don't need two parameters.

* Adding option for articles quicktask to custom admin menuitem

* Removing non-working quicktask parameter from categories view and also fixing XML structure.

* CS

* Update administrator/components/com_content/tmpl/articles/default.xml

Co-Authored-By: Quy <quy@fluxbb.org>

* CS

* Adjust language key

* Change class="switcher" to layout="joomla.form.field.radio.switcher"

Co-authored-by: Quy <quy@fluxbb.org>

SharkyKZ committed:
[4.0] Move featured logic from content table (#27891)

* Move featured logic from content table

* CS

* Add blank line

SharkyKZ committed:
Undefined methods (#27900)

chmst committed:
[4.0] fix quer for state (#27910)

infograf768 committed:
[4.0] Fix background shorthand incompatibilities (#27864)

* [4.0] Fix background shorthand incompatibilities

* rtl: Changing svg, adapting variables, changing background color

* corrected svg

* use variable

* fix RTL choices for single choice as well as multiple choices

Commit:
[4.0] Make alert inline consistent (#27707)

* Make alert inline consistent

* Additional a11y changes

* Apply a11y

* Fix conflict

Co-authored-by: Robert Deutz <rdeutz@googlemail.com>

SharkyKZ committed:
[4.0] Move namespaced extension code to subdirectory (#27687)

* Move component source to src directory

* Module module source to src directory

* Move plugin source to src directory

* Update component manifest files

* Update module manifest files

* Update plugin manifest files

* Move newly added files

brianteeman committed:
[4.0] TinyMCE new features (#27908)

With the release of tinymce 5.1 we have some extra features available. This PR adds the [toolbar drawer](https://www.tiny.cloud/docs/configure/editor-appearance/#toolbar_drawer) and the [sticky toolbar](https://www.tiny.cloud/docs/configure/editor-appearance/#toolbar_sticky)

### Sticky Toolbar
If you have a very long editor area then as you scroll the toolbar becomes sticky at the top of the screen so that it is always available

### Toolbar Drawer
This restricts the toolbar to only display one row (depending on the screen size) and if there are buttons that can not fit on the row then a `...` button appears which can be used to open and close the drawer to the other buttons.

To ensure that our `CMS Content` buttons are always displayed I moved them from the end to the beginning.

The Toolbar Drawer does not support the creation of multiple toolbars so to maintain backwards compatibility I merge the two toolbars into one.

Note: None of the presets were using the second toolbar

TinyMCE 5.2 will expand on this feature and allow greater customisation but they have not revealed how yet.

I have not made either of these as configurable settings - there are enough of those already.

mateoAdi commented:
I have tested this item :white_check_mark: successfully on c842c2a142e6d10c354d3d2c949abd45fc4cf62c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27814">issues.joomla.org/tracker/joomla-cms/27814</a>.</sub>

sakiss committed:
Merge remote-tracking branch 'origin/FinderEvents' into FinderEvents

Conflicts:
	administrator/components/com_finder/src/Model/IndexModel.php

sakiss committed:
Trigger purge event.

Removed after merge.

wilsonge commented:
This sounds sensible to me - but clearly somethings gone wrong with the rebasing here. Are you able to do this in a fresh PR please?

sakiss commented:
After the rebase issues, I made a new PR #27965  #27965
